hydra-node/src/Hydra/Crypto.hs

```diff
@@ -30,78 +40,118 @@ import Cardano.Crypto.DSIGN (
   signDSIGN,
   verifyDSIGN,
  )
-import Cardano.Crypto.Hash (Blake2b_256, Hash, castHash)
-import Cardano.Crypto.Seed (mkSeedFromBytes)
+import qualified Cardano.Crypto.DSIGN as Crypto
+import Cardano.Crypto.Hash (Blake2b_256, castHash, hashFromBytes, hashToBytes)
+import qualified Cardano.Crypto.Hash as Crypto
+import Cardano.Crypto.Seed (getSeedBytes, mkSeedFromBytes)
 import Cardano.Crypto.Util (SignableRepresentation)
 import qualified Data.Aeson as Aeson
 import qualified Data.ByteString as BS
 import qualified Data.ByteString.Base16 as Base16
 import qualified Data.Map as Map
-import Hydra.Cardano.Api (HasTypeProxy (..), SerialiseAsRawBytes (..), serialiseToRawBytesHexText)
+import Hydra.Cardano.Api (
+  AsType (AsHash, AsSigningKey, AsVerificationKey),
+  HasTextEnvelope (..),
+  HasTypeProxy (..),
+  Hash,
+  Key (..),
+  SerialiseAsCBOR,
+  SerialiseAsRawBytes (..),
+  serialiseToRawBytesHexText,
+ )
 import qualified Hydra.Contract.HeadState as OnChain
 import qualified Plutus.V2.Ledger.Api as Plutus
 import Test.QuickCheck.Instances.ByteString ()
 import Text.Show (Show (..))
 
--- | The used signature algorithm
+-- | The used hash algorithm for 'HydraKey' verification key hashes.
+type HashAlg = Blake2b_256
+
+-- | The used signature algorithm for 'HydraKey' signatures.
 type SignAlg = Ed25519DSIGN
```

Comment:
(strong :sweat_smile:) suggestion: do not use type-aliases here.

This obfuscates things, and provide little benefits. It's replacing one word by another. I would imagine having a type-class that spits back the algorithm from a given `Key HydraKey` for example if consistency is needed, but the alias makes signatures more opaque.   

Plus, this is relatively inconsistent with the rest of the Cardano codebase, which does not use type-aliases for these but rather (a) generic type parameters (.e.g `alg` -- lowercase) or (b) specialized types like `Key PaymentKey` where the hash algorithm is fixed anyway behind the scene.

Reply:
Yeah, makes sense. I was following the pattern we also used in other modules where we fixed the `Era` or `Block` types etc. I will follow your suggestion and remove this one indirection :+1: